### PR TITLE
Modify pushed flags for CLANGAUTODEBUG_BUG

### PR DESCRIPTION
**Pull request by masterleinad**
As discussed in #6699, the check in question failes when configuring with `-Werror` with the message:
```
clang-6.0: error: argument unused during compilation: '-Wa,--compress-debug-sections' [-Werror,-Wunused-command-line-argument]
```
This change seems to work for `gcc-8`, `clang-6`, `MSVC 19` and `ICC 19` while pushing `-Wno-error` fails at least for `MSVC 19`.

### CODE REVIEW

**Comment by drwells** [comment]
What a strange issue. I would normally be opposed to sending in `-g` (due to MSVC incompatibility) but since its a test for clang this should be fine.

**Comment by drwells** [comment]
I misread the code around this patch. Regardless, if it works with MSVC, then it should be fine.

I am surprised that MSVC accepts this flag at all.

**Comment by masterleinad** [comment]
> I am surprised that MSVC accepts this flag at all.

It is saying:
```
cl : Command line warning D9002 : ignoring unknown option '-g'
```

File: cmake/checks/check_01_cxx_features.cmake [hunk 1]
```diff
@@ -277,7 +277,7 @@ IF(NOT DEFINED DEAL_II_WITH_CXX14 OR DEAL_II_WITH_CXX14)
     #
     # https://llvm.org/bugs/show_bug.cgi?id=16876
     #
-    PUSH_CMAKE_REQUIRED("${DEAL_II_CXX_FLAGS_DEBUG}")
+    PUSH_CMAKE_REQUIRED("-g")
```
Comment (tamiko) [inline cmake/checks/check_01_cxx_features.cmake]: What about we simply
```
SET(_flags "${DEAL_II_CXX_FLAGS_DEBUG}")
STRIP_FLAG(_flags "-Wa,--compress-debug-sections")
PUSH_CMAKE_REQUIRED("${_flags}")
```

Comment (masterleinad) [reply cmake/checks/check_01_cxx_features.cmake]: That works as well. I'll change it.

**Pull request merged**